Source/WebCore/dom/Document.cpp

```diff
@@ -9171,6 +9174,16 @@ void Document::canvasChanged(CanvasBase& canvasBase, const std::optional<FloatRe
     }
 }
 
+void Document::updateSleepDisablerIfNeeded()
+{
+    MediaProducerMediaStateFlags activeVideoCaptureMask { MediaProducerMediaState::HasActiveVideoCaptureDevice, MediaProducerMediaState::HasActiveScreenCaptureDevice, MediaProducerMediaState::HasActiveWindowCaptureDevice };
+    if (m_mediaState & activeVideoCaptureMask) {
+        m_sleepDisabler = makeUnique<SleepDisabler>("com.apple.WebCore: Document doing camera, screen or window capture"_s, PAL::SleepDisabler::Type::Display);
+        return;
```

Comment:
Could this return early if `m_sleepDisabler` is non-null instead of recreating it?

Reply:
Good idea, will do